backend/routes/post.py

```
@@ -247,120 +264,79 @@ async def create_document_post(
+    db.add(doc_entry)
     db.commit()
-    db.refresh(new_document)
-    send_post_notifications(db, current_user, new_post)
+    db.refresh(doc_entry)
+
+    send_post_notifications(db, current_user, post)
+    return doc_entry
 
 
+@router.post("/create_text_post/", response_model=PostResponse)
+async def create_text_post(
```

Ruff (0.8.2)

Missing return type annotation for public function `create_text_post`

(ANN201)